Library/Formula/hbase.rb

```diff
@@ -4,32 +4,151 @@ class Hbase < Formula
+
+  resource "hadoop-lzo" do
+    url "https://github.com/cloudera/hadoop-lzo/archive/0.4.14.tar.gz"
+    sha256 "aa8ddbb8b3f9e1c4b8cc3523486acdb7841cd97c002a9f2959c5b320c7bb0e6c"
+  end
 
   def install
+    ENV.java_cache if build.with? "lzo"
     rm_f Dir["bin/*.cmd", "conf/*.cmd"]
     libexec.install %w[bin conf docs lib hbase-webapps]
     bin.write_exec_script Dir["#{libexec}/bin/*"]
 
-    inreplace "#{libexec}/conf/hbase-env.sh",
-      "# export JAVA_HOME=/usr/java/jdk1.6.0/",
-      "export JAVA_HOME=\"$(/usr/libexec/java_home)\""
+    if build.with? "lzo"
+      resource("hadoop-lzo").stage do
+        # Fixed upstream: https://github.com/cloudera/hadoop-lzo/blob/master/build.xml#L235
+        inreplace "build.xml",
+                  %r{(<class name="com.hadoop.compression.lzo.LzoDecompressor" />)},
+                  "\\1\n<classpath refid=\"classpath\"/>"
+        ENV["CLASSPATH"] = Dir["#{libexec}/lib/hadoop-common-*.jar"].first
+        ENV["CFLAGS"] = "-m64"
+        ENV["CXXFLAGS"] = "-m64"
+        ENV["CPPFLAGS"] = "-I/System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers"
+        system "ant", "compile-native", "tar"
+        (libexec/"lib").install Dir["build/hadoop-lzo-*/hadoop-lzo-*.jar"]
+        (libexec/"lib/native").install Dir["build/hadoop-lzo-*/lib/native/*"]
+      end
+    end
+
+    inreplace "#{libexec}/conf/hbase-env.sh" do |s|
+      # upstream bugs for ipv6 incompatibility:
+      # https://issues.apache.org/jira/browse/HADOOP-8568
+      # https://issues.apache.org/jira/browse/HADOOP-3619
+      s.gsub!("export HBASE_OPTS=\"-XX:+UseConcMarkSweepGC\"",
+              "export HBASE_OPTS=\"-Djava.net.preferIPv4Stack=true -XX:+UseConcMarkSweepGC\"")
+      s.gsub!("# export JAVA_HOME=/usr/java/jdk1.6.0/",
+              "export JAVA_HOME=\"$(/usr/libexec/java_home)\"")
+    end
+
+    # makes hbase usable out of the box
+    # upstream has been provided this patch
+    # https://issues.apache.org/jira/browse/HBASE-15426
+    inreplace "#{libexec}/conf/hbase-site.xml",
+      /<configuration>/,
+      <<-EOS.undent
+      <configuration>
+        <property>
+          <name>hbase.rootdir</name>
+          <value>#{(build.with? "hadoop") ? "hdfs://localhost:9000" : "file://"+var}/hbase</value>
+        </property>
+        <property>
+          <name>hbase.zookeeper.property.clientPort</name>
+          <value>2181</value>
+        </property>
+        <property>
+          <name>hbase.zookeeper.property.dataDir</name>
+          <value>#{var}/zookeeper</value>
+        </property>
+        <property>
+          <name>hbase.zookeeper.dns.interface</name>
+          <value>lo0</value>
+        </property>
+        <property>
+          <name>hbase.regionserver.dns.interface</name>
+          <value>lo0</value>
+        </property>
+        <property>
+          <name>hbase.master.dns.interface</name>
+          <value>lo0</value>
+        </property>
+      EOS
+
+    (var/"run/hbase").mkpath
```

Comment:
That one _should_ be fine because it's within the Cellar and consequently will exist in the bottle.